src/Storages/StorageFile.cpp

```diff
@@ -509,6 +497,27 @@ Pipe StorageFile::read(
             else
                 return metadata_snapshot->getColumns();
         };
+        // if we do multiple reads from pipe, we want to check if pipe is a regular file or a pipe
+        if (this_ptr->table_fd_was_used)
+        {
+            if (S_ISREG(this_ptr->table_fd))
+            {
+                if (this_ptr->table_fd_init_offset < 0)
+                {
+                    throw Exception("File descriptor isn't seekable, inside " + this_ptr->getName(), ErrorCodes::CANNOT_SEEK_THROUGH_FILE);
+                }
+                int seekable_table_fd = dup(this_ptr->table_fd);
```

Comment:
But why do you change the offset via dupped fd? Yes it will change the offset on the initial fd, but it will leak fd, and why do we need separate fd here?

Reply:
I thought it was incorrect to apply lseek directly to the table file descriptor, so I dupped it and lseek'd the separate fd. It seems that we don't need to do this, so `lseek(this_ptr->table_fd, this_ptr->table_fd_init_offset, SEEK_SET) < 0` will be ok.